Contest_API.md

```diff
@@ -264,6 +264,10 @@ equivalent JSON response snippets pointing to the same location:
   "href":"contests/wf14/submissions/187/files"
 ```
 
+For images in SVG format, i.e. those having a mime type of image/svg+xml,
+the actual values of `width` and `height` don't matter, only the ratio
+between them.
```

Comment:
> We have mimetypes without backquotes elsewhere in the spec so that is what I used. I can change all these if you want

Hmmm, I guess my comment was triggered in this case by the `+` in the mime type, but I think that putting all in backquotes is slightly better, to indicate that these are literal values.

Reply:
IMHO we should probably go ahead and change them all, but not on this PR. Could do just this one or cover it with the others.

Reply:
I don't really like this: "width and height don't matter, only the ratio between them" because it sounds like the sizes are completely meaningless, and that a server could feel free to change these (e.g. if the svg has an 800x600 viewbox it could expose this as width=4 and height=3).

I agree the viewbox isn't necessarily significant, but it's often the 'designed for' size. For instance, if there are two svgs with viewboxes 200x150 and 1600x1200, I would typically assume that the first is simpler and designed as an icon or small image, and the latter was designed with more detail for 2k+ displays. It's possible that a server returns both of these (or two with slightly different aspect ratios but different viewboxes), so exposing the real viewbox might be helpful.

I would prefer to just say "the width and height should be the viewbox", and if you feel we need to say more then something about how they're not necessarily significant for svgs.

Reply:
> I don't really like this: "width and height don't matter, only the ratio between them" because it sounds like the sizes are completely meaningless, and that a server could feel free to change these (e.g. if the svg has an 800x600 viewbox it could expose this as width=4 and height=3).

Well... that **is** what it means.

> I would prefer to just say "the width and height should be the viewbox", and if you feel we need to say more then something about how they're not necessarily significant for svgs.

But surely *that's* not right? The svg element has a width and a height (that defaults to `auto`, which is completely useless for us), surely that is the size that should be used (if present), since this defines the size of the viewport. the `viewbox` OTOH defines the coordinates to use inside the svg.

All that said... I don't actually fully understand how the `width` `height` and `viewbox` of an svg interact.

Reply:
I tried to read up on it, and I'm pretty sure that the viewbox (if it exists) is the internal coordinate system and defines where the image bounds are, and width/height allows you to scale that. If there's no viewbox, then the internal coordinate system is just 0,0 to width,height and matches the external size.

So maybe our implementations should look for width,height first and then viewbox width,height as backup? Either way I still think the server/spec should just directly expose what's in the SVG and let the client decide how to interpret it.

Reply:
We all agree that the clients need to know the aspect ratio of the image. That's the only thing we get with how we have currently written the spec.

You made a good point that different intended sizes might imply different level of detail. For that reason it would be useful to have some absolute size, and not only the aspect ratio. The counter argument for this though are (most) flags. E.g. the Swedish flag can be *perfectly* represented (i.e. **no** loss of detail) by a 10x16px image, so any resolution higher than that is equivalent. 

You suggested that we should simply report the actual size of the image. I agree that this would (probably) make sense for SVGs that have width and height specified in px units. But, width and height can be missing or using non px units, and then there *really* is no actual size of the image.

Reply:
If width and height of the viewbox are missing there is still the `width` and `height` attributes. If they also are missing, they are assumed to be [300 and 150](https://svgwg.org/specs/integration/#svg-css-sizing) so we should always have some value. However they can indeed be non-integer

Reply:
That's in CSS context. It's not the same thing as saying that a SVG always has a size, but rather that when it doesn't and we need one (for CSS) we'll use these instead.

Also, not only could the size be non-integer, it could be non-px. Some non-px units can be translated into px (such as in, cm, mm, pt, and pc, by assuming 96 dpi), but others depend on an outer context that does not exist for us (such as em, ex, ch, rem, vw, vh, vmin vmax, and %). Maybe there exists an default translation from this other group by assuming some default font size and viewport size?

Reply:
Maybe we should specify you can only use svgs that make sense to display in contexts that make sense (web, apps) without needing other info not present in the svg? What if an svg is 5 pixels high but 3em wide, we have no clue what to do

Reply:
I don't know how to properly specify that though. Do you? Are we sure it's actually possible?

Reply:
No, I don't know that either....

Reply:
I'm pretty sure it is possible, but this seems fairly edge case and I don't know how it would scale the internal coordinates correctly without a viewbox and preserveAspectRatio. All of the SVGs I've been testing with have had a viewbox, which seems like it should be required for reusable SVGs (although I haven't found any evidence that's required), so the simplest would be to just depend on that.

Next simplest would be to use the width/height only if they're integers (pixels). I really don't think we're going to see many SVGs that don't have a viewbox and non-pixel width/height, so IMHO it would be fine if the server just didn't expose width & height in file reference for that case.

Back to my original point: I still think the server should just expose whatever data we decide and not interpret or modify it.